utils/signer/src/lib.rs

```diff
@@ -0,0 +1,194 @@
+            let signer = Web3Signer::new(web3_provider.clone(), signer_address);
+
+            let tx = filled_tx.as_builder().unwrap().clone();
+
+            // NOTE: This is a hack because there is not a "data" field on the TransactionRequest.
+            // `eth_signTransaction` expects a "data" field with the calldata.
+            // TODO: Once alloy fixes this, we can remove this wrapper.
+            let wrapper = TransactionRequestWrapper::new(tx);
+
+            let raw: Bytes =
+                signer.provider().client().request("eth_signTransaction", (wrapper,)).await?;
+
+            let tx_envelope = TxEnvelope::decode_2718(&mut raw.as_ref()).unwrap();
+
+            let receipt = provider
+                .send_tx_envelope(tx_envelope)
+                .await
+                .context("Failed to send transaction")?
+                .with_required_confirmations(NUM_CONFIRMATIONS)
+                .with_timeout(Some(Duration::from_secs(TIMEOUT_SECONDS)))
+                .get_receipt()
+                .await?;
+
+            Ok(receipt)
+        }
+        Signer::LocalSigner(private_key) => {
+            let provider = ProviderBuilder::new()
+                .network::<Ethereum>()
+                .wallet(EthereumWallet::new(private_key.clone()))
+                .connect_http(l1_rpc);
+
+            // Set the from address to the Ethereum wallet address.
+            transaction_request.set_from(private_key.address());
+
+            // Fill the transaction request with all of the relevant gas and nonce information.
+            let filled_tx = provider.fill(transaction_request).await?;
+
+            let receipt = provider
+                .send_tx_envelope(filled_tx.as_envelope().unwrap().clone())
+                .await
+                .context("Failed to send transaction")?
+                .with_required_confirmations(NUM_CONFIRMATIONS)
+                .with_timeout(Some(Duration::from_secs(TIMEOUT_SECONDS)))
+                .get_receipt()
+                .await?;
+
+            Ok(receipt)
+        }
+    }
+}
+
+/// A wrapper around `TransactionRequest` that adds a data field as bytes.
+///
+/// This is needed because:
+/// 1. The `TransactionRequest` trait and `TransactionBuilder` trait don't include methods to set
+///    the "data" field directly (only `input()` and `set_input()` are available).
+/// 2. Web3Signer's `eth_signTransaction` method specifically expects a "data" field in the JSON-RPC
+///    request, not the "input" field.
+/// 3. While the alloy-rpc-types-eth crate has methods like `normalize_data()` and `set_both()` to
+///    work with both fields, these are only implemented on the specific Ethereum transaction type
+///    struct and aren't accessible through the generic `N::TransactionRequest` interface we're
+///    using here.
+///
+/// TODO(fakedev9999): Once alloy fixes this, we can remove this wrapper.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TransactionRequestWrapper {
```

Comment:
Is this still necessary since it's not generic on Network anymore, and TransactionRequest.input has both data and input?